apps/antalmanac/src/actions/AppStoreActions.ts

```diff
@@ -182,13 +182,18 @@ export const changeCourseColor = (sectionCode: string, term: string, newColor: s
     AppStore.changeCourseColor(sectionCode, term, newColor);
 };
 
-export const copySchedule = (to: number) => {
+export const copySchedule = (name: string, to: number) => {
     logAnalytics({
         category: analyticsEnum.addedClasses.title,
         action: analyticsEnum.addedClasses.actions.COPY_SCHEDULE,
     });
 
-    AppStore.copySchedule(to);
+    try {
+        AppStore.copySchedule(to);
+        openSnackbar('success', `Schedule copied to ${name}.`);
```

Comment:
Don't use this function.

Reply:
No, fix this now! 

Don't use the `openSnackbar` function because it's technically non-deterministic. The way it works is by extracting local component context into a global scope, which is bad because components have their own lifecycles and their local state doesn't inherently belong in global scope (this is different when using stores to share global state between components).

Prefer implementing functions with `onSuccess` and `onFailure` callbacks if you need to refer to the component's local stuff (in this case, `enqueueSnackbar` is a hook that grabs the snackbar context from the app, and it should remain local to components)

```tsx
export interface CallbackOptions {
  onSuccess: () => unknown
  onError: () => unknown
}

export const copySchedule = (name: string, to: number, options?: CallbackOptions) => {
    logAnalytics({
        category: analyticsEnum.addedClasses.title,
        action: analyticsEnum.addedClasses.actions.COPY_SCHEDULE,
    });

    // do logic

    if (success) {
      options?.onSuccess()
    } else {
      options?.onError()
    }
}
```

Any additional information needed by the component, i.e. the new schedules, messages, etc. can also be passed into the callback. It would be consumed as follows:

```tsx
function createCopyHandler(name, index, options) {
  return () => copySchedule(name, index,options)
}
function Component() {
  const { enqueueSnackbar } = useSnackbar()

  const options = useMemo(() => {
    return {
      onSuccess: () => enqueueSnackbar( ... )
    }
  }, [enqueueSnackbar])

  return (
    <MenuItem onClick={createCopyHandler(name, index, options} />
  )
}
```

Reply:
Oh that's my bad -- I thought you were talking about `.copySchedule(to)` and got mixed up. I'll take a closer look at what you wrote up later today, but at first pass it makes sense